From 2c85d2e0c376857082a918b03b90e6a480e0bf00 Mon Sep 17 00:00:00 2001
From: Jeff Trawick <trawick@apache.org>
Date: Tue, 15 Apr 2014 15:25:03 +0000
Subject: [PATCH 1/2] mod_ssl: Add hooks to allow other modules to perform
 processing at several stages of initialization and connection handling.  See
 mod_ssl_openssl.h.

This is enough to allow implementation of Certificate Transparency
outside of mod_ssl.

git-svn-id: https://svn.apache.org/repos/asf/httpd/httpd/trunk@1587607 13f79535-47bb-0310-9956-ffa450edef68

Upstream-Status: Backport [apache-httpd:6fd55ccc]

Signed-off-by: Haris Okanovic <haris.okanovic@ni.com>
---
 modules/ssl/mod_ssl.h | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/modules/ssl/mod_ssl.h b/modules/ssl/mod_ssl.h
index 0280a68..bf8eeda 100644
--- a/modules/ssl/mod_ssl.h
+++ b/modules/ssl/mod_ssl.h
@@ -29,6 +29,27 @@
 #include "httpd.h"
 #include "apr_optional.h"
 
+/* Create a set of SSL_DECLARE(type), SSL_DECLARE_NONSTD(type) and
+ * SSL_DECLARE_DATA with appropriate export and import tags for the platform
+ */
+#if !defined(WIN32)
+#define SSL_DECLARE(type)            type
+#define SSL_DECLARE_NONSTD(type)     type
+#define SSL_DECLARE_DATA
+#elif defined(SSL_DECLARE_STATIC)
+#define SSL_DECLARE(type)            type __stdcall
+#define SSL_DECLARE_NONSTD(type)     type
+#define SSL_DECLARE_DATA
+#elif defined(SSL_DECLARE_EXPORT)
+#define SSL_DECLARE(type)            __declspec(dllexport) type __stdcall
+#define SSL_DECLARE_NONSTD(type)     __declspec(dllexport) type
+#define SSL_DECLARE_DATA             __declspec(dllexport)
+#else
+#define SSL_DECLARE(type)            __declspec(dllimport) type __stdcall
+#define SSL_DECLARE_NONSTD(type)     __declspec(dllimport) type
+#define SSL_DECLARE_DATA             __declspec(dllimport)
+#endif
+
 /** The ssl_var_lookup() optional function retrieves SSL environment
  * variables. */
 APR_DECLARE_OPTIONAL_FN(char *, ssl_var_lookup,
-- 
2.6.2

